sys-power/iasl/iasl-20241212.ebuild

```
@@ -0,0 +1,113 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit edos2unix toolchain-funcs
+
+MY_PN=acpica-unix
+MY_P="${MY_PN}-${PV}"
+MY_TESTS_P="${MY_PN/ca/tests}-${PV}"
+REL_TAG="R${PV:0:4}_${PV:4:2}_${PV:6:2}"
+
+DESCRIPTION="Intel ACPI Source Language (ASL) compiler"
+HOMEPAGE="https://www.acpica.org/downloads/"
+SRC_URI="
+	https://github.com/acpica/acpica/releases/download/${REL_TAG}/${MY_P}.tar.gz
+	test? ( https://github.com/acpica/acpica/releases/download/${REL_TAG}/${MY_TESTS_P}.tar.gz )"
+S="${WORKDIR}/${MY_P}"
+
+LICENSE="iASL"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~loong ~mips ~ppc ~ppc64 ~riscv ~x86"
+IUSE="examples test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	app-alternatives/yacc
+	app-alternatives/lex"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-Makefile-fixes.patch
+	"${FILESDIR}"/${P}-Makefile-verbose.patch
+)
+
+pkg_setup() {
+	if use test && has test ${FEATURES}; then
+		ewarn 'You have selected USE="test". This will install the test results'
+		ewarn "into /usr/share/${PF}/, compressed as a tarball."
+		ewarn 'The tests themselves will only rarely die, but the test results'
+		ewarn 'are interesting for arch testing. The tests may take quite some'
+		ewarn 'time to complete.'
+	fi
+}
+
+src_compile() {
+	tc-env_build emake -C generate/unix \
+	  CC="$(tc-getCC)" \
+	  'YACC=LC_ALL=C yacc' \
+	  'YFLAGS=' \
+	  'LEX=LC_ALL=C lex' \
+	  NOFORTIFY=TRUE \
+	  NOWERROR=TRUE
+}
+
+src_test() {
+	aslts_test
+	#The aapits test currently fails, missing include probably.
+	#aapits_test
+}
+
+src_install() {
+	emake -C generate/unix install DESTDIR="${D}"
+
+	if ! use examples; then
+		rm "${ED}/usr/bin/acpiexamples" || die
+	fi
+
+	dodoc "${S}"/changes.txt
+	newdoc "${S}"/source/compiler/readme.txt compiler-readme.txt
+	newdoc "${S}"/generate/unix/readme.txt unix-readme.txt
+	newdoc "${S}"/generate/lint/readme.txt lint-readme.txt
+	newdoc "${S}"/source/compiler/new_table.txt compiler-new_table.txt
+
+	if use test && has test ${FEATURES}; then
```

The `RESTRICT` earlier in the file already ensures this:
```suggestion
	if use test; then
```